src/Carousel.tsx

```diff
@@ -487,17 +489,23 @@ function CarouselFunc(uncontrolledProps: CarouselProps, ref) {
       return undefined;
     }
 
+    let chosenInterval = interval || undefined;
+
+    if (activeIndex !== undefined && intervals.current[activeIndex]) {
+      chosenInterval = intervals.current[activeIndex];
+    }
+
     intervalHandleRef.current = window.setInterval(
       document.visibilityState ? nextWhenVisible : next,
-      interval || undefined,
+      chosenInterval,
     );
 
     return () => {
       if (intervalHandleRef.current !== null) {
         clearInterval(intervalHandleRef.current);
       }
     };
-  }, [shouldPlay, next, interval, nextWhenVisible]);
+  }, [shouldPlay, next, activeIndex, interval, nextWhenVisible]);
```

Comment:
i think we just want:

```js
const intervalRef = useCommittedRef(
  intervals.current[activeIndex] ?? interval,
)
```

using non-committed refs is unsafe (as the other carousel issue observes), and we don't want the effect hook to unnecessarily invalidate, either